drivers/can/can_common.c

```diff
@@ -160,6 +160,20 @@ static int update_sampling_pnt(uint32_t total_tq, uint32_t sample_pnt,
 		sample_pnt - sample_pnt_res;
 }
 
+/**
+ * @brief Calculate default sjw as phase_seg2 / 2 and clamp result to min/max sjw
+ *
+ * @param[in,out] res Calculated sjw is updated in the @a can_timing struct provided.
+ * @param max         Maximum timing parameters values.
+ * @param min         Minimum timing parameters values.
+ */
+static void can_calc_default_sjw(struct can_timing *res, const struct can_timing *min,
+				 const struct can_timing *max)
+{
+	res->sjw = MAX(1U, MIN(res->phase_seg1, res->phase_seg2 / 2));
```

Comment:
Do you need the MAX(1U, ...) if you CLAMP it later?
Do you have any reference for this calculation? Is it a recommendation from CiA?

Reply:
> Do you need the MAX(1U, ...) if you CLAMP it later?

Nice catch, no I don't. I added the `CLAMP()` call later, but didn't think of removing the `MAX()`. Removed now.

> Do you have any reference for this calculation? Is it a recommendation from CiA?

As far as I know, sadly the CiA does not give any recommendations regarding the SJW values, no. The only references, I have, are the ones mentioned in the issue (see https://github.com/zephyrproject-rtos/zephyr/issues/63033). The strongest argument to defaulting to `PHASE_SEG2 / 2` would be that the Linux kernel does this as well. You might want to read through their reasoning as well (again, linked from the issue).